From fc6967999b751606609ad0a40d87e78330d50e18 Mon Sep 17 00:00:00 2001
From: Lloyd Pique <lpique@google.com>
Date: Wed, 13 Sep 2023 14:18:01 -0700
Subject: [PATCH 3/6] service: Update seccomp paths

The base paths to the policy files recently changed, and one was
actually a few years out of date.

I've updated the paths to match those in
frameworks/av/media/codec2/hal/services/vendor.cpp as of commit 27a3ad1.

This fixes several post boot errors.

Bug: 300194440
Test: Manually looked at logcat errors
Change-Id: I49677fff0c1d024a9aa82d6f7d21a9c845b69bba
---
 service/service.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/service/service.cpp b/service/service.cpp
index 616e641..1fa9aae 100644
--- a/service/service.cpp
+++ b/service/service.cpp
@@ -14,13 +14,17 @@
 
 #include <v4l2_codec2/components/V4L2ComponentStore.h>
 
-// Default policy for codec2.0 service.
+// This is the absolute on-device path of the prebuild_etc module
+// "android.hardware.media.c2-default-seccomp_policy" in Android.bp.
 static constexpr char kBaseSeccompPolicyPath[] =
-        "/vendor/etc/seccomp_policy/android.hardware.media.c2@1.2-default-seccomp_policy";
+        "/vendor/etc/seccomp_policy/"
+        "android.hardware.media.c2-default-seccomp_policy";
 
-// Additional device-specific seccomp permissions can be added in this file.
+// Additional seccomp permissions can be added in this file.
+// This file does not exist by default.
 static constexpr char kExtSeccompPolicyPath[] =
-        "/vendor/etc/seccomp_policy/codec2.vendor.ext.policy";
+        "/vendor/etc/seccomp_policy/"
+        "android.hardware.media.c2-extended-seccomp_policy";
 
 int main(int /* argc */, char** /* argv */) {
     ALOGD("Service starting...");
-- 
2.39.2

